webpack.config.js

```diff
@@ -379,6 +380,112 @@ var config = {
     ],
 };
 
+async function initializeModuleFederation() {
+    function makeSingletonSharedModules(packageNames) {
+        const sharedObject = {};
+
+        for (const packageName of packageNames) {
+            const version = packageJson.dependencies[packageName];
+
+            sharedObject[packageName] = {
+                requiredVersion: version,
+                singleton: true,
+                version,
+            };
+        }
+
+        return sharedObject;
+    }
+
+    function isWebpackDevServerAvailable(baseUrl) {
+        return new Promise((resolve) => {
+            if (!DEV) {
+                resolve(false);
+                return;
+            }
+
+            const req = http.request(`${baseUrl}/remote_entry.js`, (response) => {
```

Comment:
To make it so that products are optional during development, we ping their webpack dev server on startup to see if it's running